_posts/2020-09-16-creating-a-snap-chat-mini.markdown

```diff
@@ -0,0 +1,95 @@
+
+> We recommend always waiting for the SDK initialization before rendering the app to prevent issues.
+
+### Fallback Method Calls when Working Locally
+
+When developing locally, since the SDK won't be connected to a native app, you'll have to handle some fallbacks in the places where you need to call SDK methods (when sharing to camera, chat, etc for example). If you want to test the app flow more easily instead of publishing every change to access the mini using your phone.
+
+For example, if your app uses the share-to-chat functionality, and you call the SDK method responsible for that locally, it will throw an exception. If you need to respond to that action because, for example, you need to change to a different state, you may need to implement a fallback mechanism. You can `try/catch` the exception for example, or check if `sc.app` is defined or not (in this case, `sc` is the variable containing the sdk instance and the `app` property is an object with information of the native app, that won't be populated when running locally).
+
+### Displaying a Sticker when Sharing Content Using the Camera
+
+To share content using the camera, the SDK provides a method that you can call passing a `Sticker`, this is a class defined by the SDK that contains some meta data for the sticker and the content of the actual image file base64-encoded.
+
+Depending on your design and what you want to show as the sticker, you can have the base64 string hardcoded. This may not be the best idea because it will increase the size of your JavaScript code even for users not using that functionality.
+
+For us, a more efficient solution was to use the HTML Canvas Element to draw an image only when required and get the base64 encoded image data. We had to use the full power of the canvas element to draw more things like rendering text, lines, background, etc. Then, after everything was drawn, we could get the base64 data url.
+
+If you need to implement the share-to-camera functionality, it's not easy to test the sticker generation while you are developing it. To help us create the sticker more easily, we used this simple snippet:
+
+```javascript
+// `sc` is the snap canvas SDK reference
+let base64data = generateStickerData()
+if (sc.app) {
+  // this method will prepare the sticker object and call the SDK methods
+  shareStickerUsingSDK(base64Data) 
+} else {
+  // this will show an IMG element during development locally, so we can see what we created
+  previewImage.src = base64data
+}
+```
+
+This speeds up the sticker design process a lot, since we don't need to publish the mini to test it inside Snapchat. After we were happy with the generated image, we started testing the real SDK method calls by publishing the mini to be used inside Snapchat.
```

Comment:
```suggestion
This speeds up the sticker design process a lot, since we don't need to publish the Mini to test it inside Snapchat. After we were happy with the generated image, we started testing the real SDK method calls by publishing the Mini to be used inside Snapchat.
```